zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -336,6 +328,30 @@ public void deleteProcess(final ProcessRecord processRecord) {
         processRecord.getBpmnProcessId(), processRecord.getVersion(), processRecord.getTenantId());
   }
 
+  private void invalidateCaches(
+      final String tenantId,
+      final DirectBuffer bpmnProcessId,
+      final long processDefinitionKey,
+      final long deploymentKey,
+      final int version) {
+    final var tenantIdAndProcessIdAndVersion =
+        new TenantIdAndProcessIdAndVersion(tenantId, bpmnProcessId, version);
+    processesByTenantAndProcessIdAndVersionCache.invalidate(tenantIdAndProcessIdAndVersion);
+
+    {
+      final var key = new TenantIdAndProcessDefinitionKey(tenantId, processDefinitionKey);
+      final var cached = processByTenantAndKeyCache.getIfPresent(key);
+      if (cached != null && cached.getVersion() <= version) {
+        processByTenantAndKeyCache.invalidate(key);
+      }
```

Comment:
🔧 You have wrapped this in an unnecessary code block 🙂

Reply:
I wrapped it in a new scope to keep the name of the variable short, but I will remove it :+1: